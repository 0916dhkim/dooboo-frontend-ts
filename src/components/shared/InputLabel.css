@import '../../root.css';

.root {
  padding-top: 28px;
  width: 100%;
  display: flex;
  flex-direction: column-reverse;
}

.input {
  padding-right: 12px;
  height: 44px;
  font-size: 18px;
  line-height: 27px;
  border-top: none;
  border-left: none;
  border-right: none;
  &:focus {
    outline: none;
    border-image-source: linear-gradient(to right, rgb(12, 157, 197), rgb(201, 109, 216));
    border-image-slice: 1;
  }
}

.label {
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  font-weight: normal;
}

.labelBefore {
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  font-weight: normal;
  color: #9b9b9b;
}

.labelAfter {
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  font-weight: bold;
  color: #4a4a4a;
}
