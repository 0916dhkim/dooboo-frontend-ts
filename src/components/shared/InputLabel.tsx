import React , { Component } from 'react';

const styles = require('./InputLabel.css');

export class InputLabel extends Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      inputDone: false,
    };
  }

  public onTxtChanged(e) {
    const v = e.target.value;
    if (v.length > 0) {
      this.setState({ inputDone: true });
      return;
    }
    this.setState({ inputDone: false});
  }

  public render() {
    return (
      <div className={styles.root}>
        <input
          className={styles.input}
          key={this.props.keyString}
          type={this.props.type}
          style={
            this.state.inputDone
              ? { borderBottom: '2px solid #4a4a4a' }
              : { borderBottom: '2px solid #9b9b9b' }
          }
          id={this.props.keyString}
          placeholder={this.props.placeholder}
          onChange={(e) => this.onTxtChanged(e)}
        />
        <label
          className={
            this.state.inputDone
              ? styles.labelAfter
              : styles.labelBefore
          }
        >{this.props.labelTxt}</label>
      </div>
    );
  }
}

export default InputLabel;
