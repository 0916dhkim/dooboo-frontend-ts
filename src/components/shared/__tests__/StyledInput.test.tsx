import React from 'react';
import { shallow } from 'enzyme';

import { InputLabel } from '@shared/InputLabel';

describe('InputLabel component test', () => {
  const component = shallow(
    <InputLabel
      type='email'
      test={true}
      keyString='emailInput'
      placeholder='email'
      labelTxt='email'
    />,
  );

  it('component matches with its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('the component input event is triggered with a mock argument and the state will be changed', () => {
    const input = component.find('input');
    input.simulate('change', { target: {id: 'emailInput', value: 'mock-email' }});
    expect(component.state().inputDone).toEqual(true);
  });
});
