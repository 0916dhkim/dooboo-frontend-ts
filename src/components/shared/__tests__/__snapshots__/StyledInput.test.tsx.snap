// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel component test component matches with its snapshot 1`] = `
<div
  className="profileEachCategory"
>
  <input
    className="profInput"
    id="emailInput"
    key="emailInput"
    onChange={[Function]}
    placeholder="email"
    style={
      Object {
        "borderBottom": "2px solid rgb(0, 0, 0)",
      }
    }
    type="email"
  />
  <label
    className="profCategoryTxt"
    style={
      Object {
        "color": "#000",
        "fontWeight": "bold",
      }
    }
  >
    email
  </label>
</div>
`;
